lib/windows_error/h_result/facility.rb

```diff
@@ -0,0 +1,200 @@
+module WindowsError
+  module HResult
+    module Facility
+      FacilityCode = Struct.new('FacilityCode', :name, :value, :description) do
```

Comment:
Facility codes are not error codes, so I opted to define a new class here that uses a similar interface to WindowsError::ErrorCode.

Reply:
Just curious, what's the benefit from using a `Struct` object instead of a standard `class`?

Reply:
It saves me from writing the boilerplate code in the `#initialize` method and then defining the attributes.